assembly/src/parsers/mod.rs

```diff
@@ -436,6 +453,99 @@ fn parse_constant(token: &Token) -> Result<(String, u64), ParsingError> {
     }
 }
 
+// PARSED BODY
+// ================================================================================================
+
+/// A parsed code container to bind a contiguous sequence of [Node] to their optional
+/// [SourceLocation].
+///
+/// Will yield an iterator of each [Node] with its respective [SourceLocation]. The iterator will
+/// be empty if the [SourceLocation] isn't provided.
+#[derive(Clone, Default, Eq, Debug)]
+pub struct CodeBody {
+    nodes: Vec<Node>,
+    locations: Vec<SourceLocation>,
+}
```

Comment:
Should we move this into a separate file? Maybe something like `/src/parsers/body.rs`?

Reply:
Makes sense, the struct grew quite a bit. Added @ the other PR: https://github.com/0xPolygonMiden/miden-vm/pull/903/commits/559f4301aecd59a334d8a3938ce326354c153cc0